WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/todaysstats/TodaysStatsCardViewHolder.kt

```diff
@@ -25,4 +54,80 @@ class TodaysStatsCardViewHolder(
             card.onCardClick.invoke()
         }
     }
+
+    fun TextView.updateLink(links: List<Clickable>) {
+        val spannable = SpannableString(text)
+        for (urlSpan in spannable.getSpans(0, spannable.length, URLSpan::class.java)) {
+            val startIndex = spannable.getSpanStart(urlSpan)
+            val endIndex = spannable.getSpanEnd(urlSpan)
+            links.forEach { link ->
+                spannable.withClickableSpan(startIndex, endIndex) {
+                    link.navigationAction.click()
+                }
+            }
+            spannable.withExternalLinkImageSpan(endIndex)
+            spannable.withBoldSpan(startIndex, endIndex)
+        }
+        text = spannable
+    }
+
+    private fun SpannableString.withClickableSpan(
+        startIndex: Int,
+        endIndex: Int,
+        onClickListener: (Context) -> Unit
+    ) {
+        val clickableSpan = object : ClickableSpan() {
+            override fun onClick(widget: View) {
+                widget.context?.let { onClickListener.invoke(it) }
+            }
+
+            override fun updateDrawState(ds: TextPaint) {
+                ds.color = linkColor
+                ds.typeface = Typeface.create(
+                        Typeface.DEFAULT_BOLD,
+                        Typeface.NORMAL
+                )
+                ds.isUnderlineText = false
+            }
+        }
+        setSpan(
+                clickableSpan,
+                startIndex,
+                endIndex,
+                Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
+        )
+    }
+
+    private fun SpannableString.withExternalLinkImageSpan(endIndex: Int) {
+        val drawable = ContextCompat.getDrawable(itemView.context, drawable.ic_external_white_24dp) ?: return
+        drawable.setTint(linkColor)
+        drawable.setBounds(
+                EXTERNAL_LINK_DRAWABLE_BOUND_LEFT,
+                0,
+                drawable.intrinsicWidth / 2,
+                drawable.intrinsicHeight / 2
+        )
+        setSpan(
+                ImageSpan(drawable, ImageSpan.ALIGN_BASELINE),
+                endIndex,
+                endIndex + 1,
```

Comment:
This specific line:
 - Made the code crash with an out-of-bounds error in locales/translations where there were no character after the link.
 - "Ate" the character just after the link in other locales which had one, by replacing it with the drawable instead —e.g. in English, it ate the `.` after the link
 - This issue was even more impactful in locales using ideograms (e.g. Korean, Japanese, Chinese), as by "eating" the next character after the link, that replace a whole ideogram… rendering the translation nonsensical

See some more info and context in [this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/16446#discussion_r862026952)